pkg/solver/server.go

```diff
@@ -62,17 +63,44 @@ func NewSolverServer(
  *
 */
 
+func exemptIPKeyFunc(exemptIPs []string) func(r *corehttp.Request) (string, error) {
+	return func(r *corehttp.Request) (string, error) {
+		ip, _, err := net.SplitHostPort(r.RemoteAddr)
+		if err != nil {
+			ip = r.RemoteAddr
+		}
+		
+
+		for _, exemptIP := range exemptIPs {
+			if ip == exemptIP {
+				return "", nil
+			}
+		}
+		
+		return ip, nil
+	}
+}
+
```

Comment:
Could we move this helper down below `disconnectCB` to keep it with the the other helpers?